src/librustc_save_analysis/dump.rs

```diff
@@ -8,33 +8,31 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-use syntax::codemap::Span;
-
 use super::data::*;
 
 pub trait Dump {
-    fn crate_prelude(&mut self, _: Span, _: CratePreludeData) {}
-    fn enum_data(&mut self, _: Span, _: EnumData) {}
-    fn extern_crate(&mut self, _: Span, _: ExternCrateData) {}
-    fn impl_data(&mut self, _: Span, _: ImplData) {}
-    fn inheritance(&mut self, _: InheritanceData) {}
-    fn function(&mut self, _: Span, _: FunctionData) {}
-    fn function_ref(&mut self, _: Span, _: FunctionRefData) {}
-    fn function_call(&mut self, _: Span, _: FunctionCallData) {}
-    fn method(&mut self, _: Span, _: MethodData) {}
-    fn method_call(&mut self, _: Span, _: MethodCallData) {}
-    fn macro_data(&mut self, _: Span, _: MacroData) {}
-    fn macro_use(&mut self, _: Span, _: MacroUseData) {}
-    fn mod_data(&mut self, _: ModData) {}
-    fn mod_ref(&mut self, _: Span, _: ModRefData) {}
-    fn struct_data(&mut self, _: Span, _: StructData) {}
-    fn struct_variant(&mut self, _: Span, _: StructVariantData) {}
-    fn trait_data(&mut self, _: Span, _: TraitData) {}
-    fn tuple_variant(&mut self, _: Span, _: TupleVariantData) {}
-    fn type_ref(&mut self, _: Span, _: TypeRefData) {}
-    fn typedef(&mut self, _: Span, _: TypedefData) {}
-    fn use_data(&mut self, _: Span, _: UseData) {}
-    fn use_glob(&mut self, _: Span, _: UseGlobData) {}
-    fn variable(&mut self, _: Span, _: VariableData) {}
-    fn variable_ref(&mut self, _: Span, _: VariableRefData) {}
+    fn crate_prelude(&mut self, CratePreludeData);
+    fn enum_data(&mut self, EnumData);
+    fn extern_crate(&mut self, ExternCrateData);
+    fn impl_data(&mut self, ImplData);
+    fn inheritance(&mut self, InheritanceData);
+    fn function(&mut self, FunctionData);
+    fn function_ref(&mut self, FunctionRefData);
+    fn function_call(&mut self, FunctionCallData);
+    fn method(&mut self, MethodData);
+    fn method_call(&mut self, MethodCallData);
+    fn macro_data(&mut self, MacroData);
+    fn macro_use(&mut self, MacroUseData);
+    fn mod_data(&mut self, ModData);
+    fn mod_ref(&mut self, ModRefData);
+    fn struct_data(&mut self, StructData);
+    fn struct_variant(&mut self, StructVariantData);
+    fn trait_data(&mut self, TraitData);
+    fn tuple_variant(&mut self, TupleVariantData);
+    fn type_ref(&mut self, TypeRefData);
+    fn typedef(&mut self, TypedefData);
+    fn use_data(&mut self, UseData);
+    fn use_glob(&mut self, UseGlobData);
+    fn variable(&mut self, VariableData);
+    fn variable_ref(&mut self, VariableRefData);
```

Comment:
Methods which are empty by default are trivial to implement, so there is not much advantage to not having to write them out. On the other hand, if we have defaults and we add another, then any implementers will be silently broken. With required methods, the compiler tells us we have to fix the `impl`s.

On the other hand, I can see that if you just want (e.g.) globs, then it is painful to write out all the other methods. I guess Rust wants more nuanced inheritance for this kind of thing. I'll revert this change.